### ISSUE TITLE
Installing via Brew fails on macOS Monteray

### ISSUE THREAD

**Issue by iharkatkavets** [Snaipe/Criterion#391]
I try to install the library via `brew` using command `brew install snaipe/soft/criterion` 

But installing fails with error.  Please see full output below

```bash
kotkovetsiv@macbook Subarray Division $ brew install snaipe/soft/criterion
Updating Homebrew...
==> Auto-updated Homebrew!
Updated 2 taps (homebrew/core and homebrew/cask).
==> Updated Formulae
Updated 2 formulae.
==> Updated Casks
Updated 1 cask.

==> Downloading https://ghcr.io/v2/homebrew/core/nanomsg/manifests/1.1.5
Already downloaded: /Users/kotkovetsiv/Library/Caches/Homebrew/downloads/b9e922ac74ec517db78d67ef92a74caeb60292891d01eab582e4b896a0a721c3--nanomsg-1.1.5.bottle_manifest.json
==> Downloading https://ghcr.io/v2/homebrew/core/nanomsg/blobs/sha256:3f59f226fe111172467d9d7d187aa486659e5b58abc3571b6612b544d4dc43e9
Already downloaded: /Users/kotkovetsiv/Library/Caches/Homebrew/downloads/a89646d0aff533b6758203cc05097506af95dde0ed53b238e393fc39fb57d6ff--nanomsg--1.1.5.monterey.bottle.tar.gz
==> Downloading https://ghcr.io/v2/homebrew/core/cmake/manifests/3.21.4
Already downloaded: /Users/kotkovetsiv/Library/Caches/Homebrew/downloads/b7ef8d6eb909e967d072212c62e71bfb8e94e6227ae2d3567bbabcc561fd9fff--cmake-3.21.4.bottle_manifest.json
==> Downloading https://ghcr.io/v2/homebrew/core/cmake/blobs/sha256:d6fb0df5818454f9f9074552eb06c34146da91c5212c16057b544a3677a29d9a
Already downloaded: /Users/kotkovetsiv/Library/Caches/Homebrew/downloads/aa67f42e613f2d2295dd9d2722cad83179b924e4877cbbb3f1aff5b1d470a7e5--cmake--3.21.4.monterey.bottle.tar.gz
==> Cloning https://github.com/Snaipe/Criterion.git
Updating /Users/kotkovetsiv/Library/Caches/Homebrew/criterion--git
==> Checking out tag v2.3.3
HEAD is now at a64b860 v2.3.3: merging fix/247.
HEAD is now at a64b860 v2.3.3: merging fix/247.
Entering 'dependencies/debugbreak'
Entering 'dependencies/klib'
Entering 'dependencies/nanopb'
/Users/kotkovetsiv/Library/Caches/Homebrew/criterion--git/dependencies/debugbreak
/Users/kotkovetsiv/Library/Caches/Homebrew/criterion--git/dependencies/klib
/Users/kotkovetsiv/Library/Caches/Homebrew/criterion--git/dependencies/nanopb
==> Installing criterion from snaipe/soft
==> Installing dependencies for snaipe/soft/criterion: nanomsg and cmake
==> Installing snaipe/soft/criterion dependency: nanomsg
==> Pouring nanomsg--1.1.5.monterey.bottle.tar.gz
🍺  /usr/local/Cellar/nanomsg/1.1.5: 25 files, 348.8KB
==> Installing snaipe/soft/criterion dependency: cmake
==> Pouring cmake--3.21.4.monterey.bottle.tar.gz
🍺  /usr/local/Cellar/cmake/3.21.4: 2,953 files, 42.6MB
==> Installing snaipe/soft/criterion
==> cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo -DI18N=OFF .
==> make install
Last 15 lines from /Users/kotkovetsiv/Library/Logs/Homebrew/criterion/02.make:
[ 86%] Building C object CMakeFiles/criterion.dir/src/protocol/protocol.c.o
[ 88%] Building C object CMakeFiles/criterion.dir/src/protocol/messages.c.o
[ 89%] Building C object CMakeFiles/criterion.dir/src/protocol/connect.c.o
[ 91%] Building C object CMakeFiles/criterion.dir/src/string/extglobmatch.c.o
[ 92%] Building C object CMakeFiles/criterion.dir/src/string/brz.c.o
[ 94%] Building C object CMakeFiles/criterion.dir/dependencies/nanopb/pb_common.c.o
[ 95%] Building C object CMakeFiles/criterion.dir/dependencies/nanopb/pb_encode.c.o
[ 97%] Building C object CMakeFiles/criterion.dir/dependencies/nanopb/pb_decode.c.o
[ 98%] Building C object CMakeFiles/criterion.dir/src/core/theories.c.o
[100%] Linking C shared library libcriterion.dylib
ld: unaligned pointer(s) for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [libcriterion.3.1.0.dylib] Error 1
make[1]: *** [CMakeFiles/criterion.dir/all] Error 2
make: *** [all] Error 2

If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):
  https://github.com/snaipe/homebrew-soft/issues

These open issues may also help:
linker error macOS Mojave 10.14.1 installing Criterion https://github.com/Snaipe/homebrew-soft/issues/3
```

Does anybody know how to fix the issue? Thank you in advance!

**Comment by restocked** [comment]
Same issue is happening with a different architecture, I tested on M1

**Comment by MrAnno** [comment]
The verbose output is the following. It seems the unaligned pointers are coming from nanopb-generated units.

```
ld: warning: pointer not aligned at address 0x2C57A (_criterion_protocol_birth_msg + 10 from src/libcriterion.3.dylib.p/protocol_criterion.pb.c.o)
ld: warning: pointer not aligned at address 0x2C572 (_criterion_protocol_birth_msg + 2 from src/libcriterion.3.dylib.p/protocol_criterion.pb.c.o)
ld: warning: pointer not aligned at address 0x2C5BE (_criterion_protocol_death_msg + 10 from src/libcriterion.3.dylib.p/protocol_criterion.pb.c.o)
ld: warning: pointer not aligned at address 0x2C5B6 (_criterion_protocol_death_msg + 2 from src/libcriterion.3.dylib.p/protocol_criterion.pb.c.o)
ld: warning: pointer not aligned at address 0x2C602 (_criterion_protocol_param_entry_msg + 10 from src/libcriterion.3.dylib.p/protocol_criterion.pb.c.o)
ld: warning: pointer not aligned at address 0x2C5FA (_criterion_protocol_param_entry_msg + 2 from src/libcriterion.3.dylib.p/protocol_criterion.pb.c.o)
ld: warning: pointer not aligned at address 0x2C63A (_criterion_protocol_param_list_msg + 10 from src/libcriterion.3.dylib.p/protocol_criterion.pb.c.o)
ld: warning: pointer not aligned at address 0x2C632 (_criterion_protocol_param_list_msg + 2 from src/libcriterion.3.dylib.p/protocol_criterion.pb.c.o)
ld: warning: pointer not aligned at address 0x2C67A (_criterion_protocol_result_msg + 10 from src/libcriterion.3.dylib.p/protocol_criterion.pb.c.o)
ld: warning: pointer not aligned at address 0x2C672 (_criterion_protocol_result_msg + 2 from src/libcriterion.3.dylib.p/protocol_criterion.pb.c.o)
ld: warning: pointer not aligned at address 0x2C6BA (_criterion_protocol_assert_msg + 10 from src/libcriterion.3.dylib.p/protocol_criterion.pb.c.o)
ld: warning: pointer not aligned at address 0x2C6B2 (_criterion_protocol_assert_msg + 2 from src/libcriterion.3.dylib.p/protocol_criterion.pb.c.o)
ld: warning: pointer not aligned at address 0x2C6FE (_criterion_protocol_ack_msg + 10 from src/libcriterion.3.dylib.p/protocol_criterion.pb.c.o)
ld: warning: pointer not aligned at address 0x2C6F6 (_criterion_protocol_ack_msg + 2 from src/libcriterion.3.dylib.p/protocol_criterion.pb.c.o)
ld: warning: pointer not aligned at address 0x2C762 (_criterion_protocol_submessage_msg + 10 from src/libcriterion.3.dylib.p/protocol_criterion.pb.c.o)
ld: warning: pointer not aligned at address 0x2C75A (_criterion_protocol_submessage_msg + 2 from src/libcriterion.3.dylib.p/protocol_criterion.pb.c.o)
ld: warning: pointer not aligned at address 0x2C7A2 (_criterion_protocol_msg_msg + 18 from src/libcriterion.3.dylib.p/protocol_criterion.pb.c.o)
ld: warning: pointer not aligned at address 0x2C79A (_criterion_protocol_msg_msg + 10 from src/libcriterion.3.dylib.p/protocol_criterion.pb.c.o)
ld: warning: pointer not aligned at address 0x2C792 (_criterion_protocol_msg_msg + 2 from src/libcriterion.3.dylib.p/protocol_criterion.pb.c.o)
ld: unaligned pointer(s) for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by Snaipe** [comment]
Good catch. Seems to be a known issue: https://github.com/nanopb/nanopb/issues/254

TLDR; we probably ought to #define PB_NO_PACKED_STRUCTS.

**Comment by MrAnno** [comment]
Upgrading to nanopb-0.4.5 seems to fix it as well.

I'll open a pull request soon.

**Comment by MrAnno** [comment]
A fresh build from the `bleeding` branch can be installed with
```
brew install mranno/tap/criterion
```
